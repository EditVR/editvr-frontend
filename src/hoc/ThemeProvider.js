/**
 * @file ThemeProvider.js
 * Exports a HOC ThemeProvider component that attaches children their MUI theme.
 */

import React from 'react';
import { MuiThemeProvider } from '@material-ui/core';

import muiTheme from '../styles/muiTheme';

const ThemeProvider = WrappedComponent => props => (
  <MuiThemeProvider theme={muiTheme}>
    <WrappedComponent {...props} />
  </MuiThemeProvider>
);

export default ThemeProvider;
