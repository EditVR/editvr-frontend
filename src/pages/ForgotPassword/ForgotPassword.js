/**
 * @file ForgotPassword.js
 * Exports a React component that renders EditVR's reset password page.
 */

import React from 'react';

import { ForgotPasswordForm } from '../../components';
import { ThinLayout } from '../../layouts';

const ForgotPassword = () => (
  <ThinLayout title="Reset your EditVR Account Password">
    <ForgotPasswordForm />
  </ThinLayout>
);

export default ForgotPassword;
