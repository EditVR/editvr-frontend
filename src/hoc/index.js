/**
 * @file index.js
 * Exports all hoc components.
 */

import ThemeProvider from './ThemeProvider';

export { ThemeProvider };
