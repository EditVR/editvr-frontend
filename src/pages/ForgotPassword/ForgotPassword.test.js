/**
 * @file Register.test.js
 * Contains tests for Register.js.
 */

import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';

import { ThemeProvider } from '../../hoc';
import ForgotPassword from './ForgotPassword';

const ThemedForgotPassword = ThemeProvider(ForgotPassword);

describe('<ForgotPassword />', () => {
  it('Matches its snapshot', () => {
    const store = configureStore()({
      loadingBar: {
        default: 0
      },
      user: {
        error: null
      }
    });

    expect(
      renderer
        .create(
          <Provider store={store}>
            <Router>
              <ThemedForgotPassword />
            </Router>
          </Provider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
