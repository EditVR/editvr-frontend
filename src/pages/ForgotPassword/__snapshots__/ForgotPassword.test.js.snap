// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword /> Matches its snapshot 1`] = `
<div
  className="ThinLayout-wrapper-1"
  id="layout__wrapper"
>
  <div />
  <div
    className="ThinLayout-thinWrapper-2"
    id="layout__thin-wrapper"
  >
    <header
      className="Header-wrapper-3"
    >
      <img
        alt="EditVR logo"
        className="Header-logo-4"
        src="editvr-logo.svg"
      />
    </header>
    <h1
      className="MuiTypography-root-5 MuiTypography-headline-10"
    >
      Reset your EditVR Account Password
    </h1>
    <form
      onSubmit={[Function]}
    >
      <p
        className="MuiTypography-root-5 MuiTypography-body1-14"
        type="body1"
      >
        To re-set your password, please enter the email associated with your EditVR account in the form below, and click Reset Password. Once you do this, you will be redirected to the Login page, and an email will be sent to your email address with further instructions.
      </p>
      <div
        aria-describedby="email-helper-text"
        className="MuiFormControl-root-31 C-textField-29"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <label
          className="MuiFormLabel-root-40 MuiInputLabel-root-35 MuiInputLabel-formControl-36 MuiInputLabel-animated-39"
          data-shrink={false}
          htmlFor="email"
        >
          Email
          <span
            className="MuiFormLabel-asterisk-44"
          >
            â€‰*
          </span>
        </label>
        <div
          className="MuiInput-root-45 MuiInput-formControl-46 MuiInput-underline-49"
        >
          <input
            aria-invalid={false}
            aria-required={true}
            autoComplete={undefined}
            autoFocus={undefined}
            className="MuiInput-input-53 MuiInput-inputType-56"
            defaultValue={undefined}
            disabled={false}
            id="email"
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={true}
            rows={undefined}
            type="email"
            value={undefined}
          />
        </div>
        <p
          className="MuiFormHelperText-root-58"
          id="email-helper-text"
        >
          Enter the email address for your account.
        </p>
      </div>
      <button
        className="MuiButtonBase-root-78 MuiButton-root-62 MuiButton-raised-68 MuiButton-raisedPrimary-69 C-button-30"
        disabled={false}
        onBlur={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="submit"
      >
        <span
          className="MuiButton-label-63"
        >
          Reset Password
        </span>
        <span
          className="MuiTouchRipple-root-81"
        />
      </button>
    </form>
    <footer
      className="Footer-wrapper-88"
    >
      <div
        className="Footer-elements-89"
      >
        <a
          href="https://fourkitchens.com"
        >
          <p
            className="MuiTypography-root-5 MuiTypography-body1-14 Footer-text-91"
            type="body1"
          >
            EditVR is a product of
          </p>
          <img
            alt="Four Kitchens"
            className="Footer-logo-90"
            src="4k-logo-reversed.svg"
          />
        </a>
      </div>
    </footer>
  </div>
</div>
`;
